enhancements/operator-conditions-lib.md

```diff
@@ -49,40 +49,111 @@ Operators need to be able to read the Status CR created by OLM, and also have th
+```go
+// Option is a function that applies a change to a condition.
+// This can be used to set optional condition fields, like reasons
+// and messages.
+type Option func(*metav1.Condition)
+
+func WithReason(reason string) Option {
+	return func(c *metav1.Condition) {
+		c.Reason = reason
+	}
+}
+
+func WithMessage(message string) Option {
+	return func(c *metav1.Condition) {
+		c.Message = message
+	}
+}
+```
 
-2. Specify conditions in the CRD:
+The library will provide helpers to return a new `Conditions` interface when a controller-runtime client and the `conditionType` is provided.
 
-The operators can introduce OLM-supported conditions which influence OLM's behavior or can also specify any custom ([non-OLM supported][non-olm-supported-condition]) condition for communicating the status of the operator to users. An example Status CR is given below:
+```go
 
-```yaml
-apiVersion: operators.coreos.com/v1
-kind: Status
-metadata:
-  name: foo-operator-zdgen-asdf # RandomGen Name
-  namespace: operators
-status:
-  conditions:
-    type: Upgradeable # The name of the `foo` OLM Supported Condition.
-    status: "False"   # The operator is not ready to be upgraded.
-    reason: "migration"
-    message: "The operator is performing a migration."
-    lastTransitionTime: "2020-08-24T23:15:55Z"
+// conditionAccessor contains the controller-runtime client and the namespacedName
+// required to access the objects on cluster.
+type conditionAccessor struct {
+  client Client.client
+  namespacedName types.NamespacedName
+}
+
+// condition contains the conditionAccessor and the conditionType which
+// needs to be added/modified.
+type condition struct {
+  conditions conditionAccessor
+  conditionType metav1.ConditionType
+}
+
+// condition will implement the Set and Get methods defined in Condition interface.
+var _ Condition = &condition{}
+
+// NewCondition will return a new Condition to access the condition of the specified
+// conditionType from the CR using the client provided by the user. This method will
+// internally obtain the name of the condition from the environment variable set by OLM and the
+// namespace from the associated service account secret.
+func NewCondition(cl client.Client, conditionType metav1.ConditionType) (Condition, error) {
+  // return a new conditionAccessor with the client provided by the user and the namespacedName
+  // obtained from the env variables. An error is returned if the namespacedName cannot be found.
+  c, err := NewConditions(cl)
```

Comment:
Non-blocking implementation nit: this could probably be named better (address in impl PR).

Reply:
Yeah, will do in the implementation PR